Lib/test/test_time.py

```diff
@@ -312,7 +314,13 @@ def test_ctime(self):
         self.assertEqual(time.ctime(t), 'Sun Sep 16 01:03:52 1973')
         t = time.mktime((2000, 1, 1, 0, 0, 0, 0, 0, -1))
         self.assertEqual(time.ctime(t), 'Sat Jan  1 00:00:00 2000')
-        for year in [-100, 100, 1000, 2000, 2050, 10000]:
+        # the largest year on VxWorks is 2038
```

Comment:
I don't understand why you skip year 2050 on VxWorks. What is the behavior? Does it crash? If yes, I would prefer to fix the C code rather than skipping the test.

For example, raise an exception (only on VxWorks) if year if larger than 2038?

Something like:

```
#ifdef __VXWORKS__
    if (buf.tm_year + 1900 < 1 || 2038 < buf.tm_year + 1900) {
        PyErr_SetString(PyExc_ValueError,
                        "strftime() requires year in [1; 2038]");
```

Reply:
ping @pxinwr? Can you try the proposed change in timemodule.c instead?

Reply:
@vstinner Sorry for late response. I will fix this soon.